wit/timezone.wit

```diff
@@ -3,53 +3,48 @@ package wasi:clocks@0.2.0;
+    /// Return information needed to display the given `instant`. This includes
+    /// the UTC offset and the time zone name.
     ///
-    /// If the timezone cannot be determined for the given `datetime`, return a
-    /// `timezone-display` for `UTC` with a `utc-offset` of 0 and no daylight
-    /// saving time.
+    /// If the timezone cannot be determined for the given `instant`, return a
+    /// `timezone-display` for `UTC` with a `utc-offset` of 0.
     @unstable(feature = clocks-timezone)
-    display: func(when: datetime) -> timezone-display;
+    display: func(when: instant) -> timezone-display;
 
     /// The same as `display`, but only return the UTC offset.
     @unstable(feature = clocks-timezone)
-    utc-offset: func(when: datetime) -> s32;
+    utc-offset: func(when: instant) -> s64;
 
-    /// Information useful for displaying the timezone of a specific `datetime`.
+    /// Information useful for displaying the timezone of a specific `instant`.
     ///
     /// This information may vary within a single `timezone` to reflect daylight
     /// saving time adjustments.
     @unstable(feature = clocks-timezone)
     record timezone-display {
-        /// The number of seconds difference between UTC time and the local
+        /// The number of nanoseconds difference between UTC time and the local
         /// time of the timezone.
         ///
-        /// The returned value will always be less than 86400 which is the
-        /// number of seconds in a day (24*60*60).
+        /// The returned value will always be less than 86,400,000,000,000 which
+        /// is the number of nanoseconds in a day (24*60*60*1e9).
         ///
         /// In implementations that do not expose an actual time zone, this
         /// should return 0.
-        utc-offset: s32,
+        utc-offset: s64,
 
-        /// The abbreviated name of the timezone to display to a user. The name
-        /// `UTC` indicates Coordinated Universal Time. Otherwise, this should
-        /// reference local standards for the name of the time zone.
+        /// The IANA identifier of the timezone. The id `UTC` indicates
+        /// Coordinated Universal Time. Otherwise, this should be an identifier
+        /// from the IANA Time Zone Database.
+        ///
+        /// For displaying to a user, the identifier should be converted into a
+        /// localized name by means of an internationalization API.
         ///
         /// In implementations that do not expose an actual time zone, this
         /// should be the string `UTC`.
         ///
         /// In time zones that do not have an applicable name, a formatted
         /// representation of the UTC offset may be returned, such as `-04:00`.
-        name: string,
-
-        /// Whether daylight saving time is active.
-        ///
-        /// In implementations that do not expose an actual time zone, this
-        /// should return false.
-        in-daylight-saving-time: bool,
+        id: string,
```

Comment:
Does an IANA timezone IDs change for a given user for different instants in time? If not, maybe this part should be removed from this function and put somewhere else (probably a `func() -> string`) that does not require an `instant` as input.

E.g., if someone's time zone is currently `Asia/Kolkata`, is that still the case if asking about a date-time in 1900, or would that be `Asia/Calcutta` instead? Or, similarly for time zones which have split or joined at various points in history.

Reply:
Good point, there isn't any need for this to be instant-dependent. We could probably go a bit farther and refactor the interface because `timezone-display` makes less sense without the DST flag. I'll add a commit with how that might look.